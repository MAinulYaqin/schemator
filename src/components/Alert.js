import React from "react";
import PropTypes from "prop-types";
import Modal from "react-modal";
import styled from "styled-components";
import { connect } from "react-redux";

import "typeface-roboto";

import { setAlert } from "../store/actions";

const Row = styled.div`
  display: flex;
`;

const Message = styled.p`
  font-family: Roboto;
  font-size: 16px;
  margin: 0px;
`;

const ButtonWrapper = styled.div`
  text-align: center;
`;

const Button = styled.button`
  width: 96px;
  background: #555;
  color: white;
  padding: 8px;
  border: 0px;
  border-radius: 4px;
  font-family: Roboto;
  font-size: 16px;
  margin-top: 24px;
  outline: none;
  align-self: center;

  &:hover {
    background: #444;
  }
`;

export const Alert = ({ isOpen, message, icon, iconColor, showAlert }) => {
  const Icon = styled(icon)`
    font-size: 48px;
    color: ${({ iconColor }) => iconColor};
    margin-right: 24px;
  `;

  const closeAlert = () => showAlert({ isOpen: false });

  return (
    <Modal
      style={{
        overlay: {
          background: "rgba(0, 0, 0, 0.5)"
        },
        content: {
          position: "absolute",
          top: "50%",
          left: "50%",
          transform: "translate(-50%, -50%)",
          width: 320,
          height: 110,
          padding: 24,
          background: "white"
        }
      }}
      isOpen={isOpen}
      onRequestClose={closeAlert}
    >
      <Row>
        <Icon iconColor={iconColor} />
        <Message>{message}</Message>
      </Row>
      <ButtonWrapper>
        <Button onClick={closeAlert}>OK</Button>
      </ButtonWrapper>
    </Modal>
  );
};

Alert.propTypes = {
  isOpen: PropTypes.bool,
  message: PropTypes.string,
  icon: PropTypes.func,
  iconColor: PropTypes.string,
  showAlert: PropTypes.func
};

const mapStateToProps = ({ alert }) => ({ ...alert });

const mapDispatchToProps = dispatch => ({
  showAlert: options => dispatch(setAlert(options))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Alert);
