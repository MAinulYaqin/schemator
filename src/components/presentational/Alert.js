import React from "react";
import PropTypes from "prop-types";
import Modal from "react-modal";
import styled from "styled-components";

import "typeface-roboto";

const Row = styled.div`
  display: flex;
`;

const Message = styled.p`
  font-family: Roboto;
  font-size: 16px;
  margin: 0px;
`;

const ButtonWrapper = styled.div`
  text-align: center;
`;

export const Button = styled.button`
  width: 96px;
  background: #555;
  color: white;
  padding: 8px;
  border: 0px;
  border-radius: 4px;
  font-family: Roboto;
  font-size: 16px;
  margin-top: 24px;
  outline: none;
  align-self: center;

  &:hover {
    background: #444;
  }
`;

const Alert = ({
  isOpen,
  message,
  icon,
  iconColor,
  onRequestClose,
  onClickOK
}) => {
  const Icon = styled(icon)`
    font-size: 48px;
    margin-right: 24px;
  `;

  return (
    <Modal
      style={{
        overlay: {
          background: "rgba(0, 0, 0, 0.5)"
        },
        content: {
          position: "absolute",
          top: "50%",
          left: "50%",
          transform: "translate(-50%, -50%)",
          width: 320,
          height: 110,
          padding: 24,
          background: "white"
        }
      }}
      isOpen={isOpen}
      onRequestClose={onRequestClose}
    >
      <Row>
        <Icon color={iconColor} />
        <Message>{message}</Message>
      </Row>
      <ButtonWrapper>
        <Button onClick={onClickOK}>OK</Button>
      </ButtonWrapper>
    </Modal>
  );
};

Alert.propTypes = {
  isOpen: PropTypes.bool,
  message: PropTypes.string,
  icon: PropTypes.func,
  iconColor: PropTypes.string,
  onRequestClose: PropTypes.func,
  onClickOK: PropTypes.func
};

export default Alert;
