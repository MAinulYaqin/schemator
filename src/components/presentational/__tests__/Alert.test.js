import React from "react";
import { shallow } from "enzyme";
import MdInfo from "react-icons/lib/md/info";

import Alert, { Button } from "../Alert";

describe("Alert", () => {
  const setup = propOverrides => {
    const props = {
      isOpen: false,
      iconColor: "black",
      message: "My message",
      icon: MdInfo,
      onClickOK: jest.fn(),
      onRequestClose: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Alert {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClickOK' while OK button clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(Button).simulate("click");

    expect(props.onClickOK).toBeCalled();
  });
});
