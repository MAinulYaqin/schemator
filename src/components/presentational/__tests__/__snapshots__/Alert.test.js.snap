// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should renders default 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "background": "white",
        "height": 110,
        "left": "50%",
        "padding": 24,
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": 320,
      },
      "overlay": Object {
        "background": "rgba(0, 0, 0, 0.5)",
      },
    }
  }
>
  <styled.div>
    <Styled(MdInfo)
      color="black"
    />
    <styled.p>
      My message
    </styled.p>
  </styled.div>
  <styled.div>
    <styled.button
      onClick={[Function]}
    >
      OK
    </styled.button>
  </styled.div>
</Modal>
`;
